File: online/LeetCodeCN/1~100/70.kt
class Solution {
    var dp: MutableMap<Int, Int> = mutableMapOf(1 to 1, 2 to 2)

    fun climbStairs(n: Int): Int {
        var ans: Int? = dp[n]

        if (ans == null) {
            ans = climbStairs(n-1) + climbStairs(n-2)
            dp[n] = ans
        }

        return ans
    }
}

File: online/LeetCodeCN/1~100/70.ts
let dp = new Map<number, number>([[1, 1], [2, 2]]);

function climbStairs(n: number): number {
    let ans = dp.get(n);

    if (!ans) {
        ans = climbStairs(n-1) + climbStairs(n-2);
        dp.set(n, ans);
    }

    return ans;
};
